### ISSUE TITLE
writing new rows as part of the same query gets exponentially slower over time

### ISSUE THREAD

**Issue by mbyio** [xo/usql#66]
While connected to a MySQL database, I pasted 500 lines that were each 53 characters long into usql as part of one big query. I could actually see usql get slower and slower as it read new rows from stdin. I also couldn't abort using ctrl-c, though that seems like a separate issue. It started out reading quickly but at around 400 rows in it was doing to reading only about 1 row per second. It seems like there's an n^2 algorithm somewhere based on the number of rows or the total size of a query.

Obviously copying and pasting scripts into usql is not the most efficient way to do things, but still it should not be this slow, and I think most people would expect it to work.

This should be with the latest usql; I just installed it today with `go get`.

**Comment by kenshaw** [comment]
Yes, it's actually more like a `n^4` algorithm, if you had syntax highlighting enabled. The current way syntax highlighting is done is a hack necessitated by the use of two underlying packages, `chroma` and `readline`, both of which would need to be heavily refactored (mostly `readline` is the culprit here) in order to fix performance issues.

Additionally, the issue is that when you copy/paste into `usql`, however it works with `readline` it gets inserted as a single entry. When I wrote the part of `usql` that interfaces with `readline`, I did not know/expect this behavior.

For ~500 lines, I would also recommend trying to use `\i` or `\ir` at least temporarily. This issue (overall) will be fixed before 1.0, which I hope to have done before the end of the year.

*tl;dr:* this is known, will be fixed, but due to complexity, can't be easily fixed

**Comment by jcarnu** [comment]
Same behaviour with 140 SQL line query. Understood it's known fact and that it will be fixed. Just saying it'll be appreciated (as it is for the rest of the nice features usql already embeds). Thanks.
